Binary files linux-5.10.y-a/._arch and linux-5.10.y-b/._arch differ
Binary files linux-5.10.y-a/arch/arm64/boot/._.DS_Store and linux-5.10.y-b/arch/arm64/boot/._.DS_Store differ
Binary files linux-5.10.y-a/arch/arm64/boot/.DS_Store and linux-5.10.y-b/arch/arm64/boot/.DS_Store differ
Binary files linux-5.10.y-a/arch/arm64/boot/dts/amlogic/._meson-axg.dtsi and linux-5.10.y-b/arch/arm64/boot/dts/amlogic/._meson-axg.dtsi differ
diff -urN linux-5.10.y-a/arch/arm64/boot/dts/amlogic/meson-axg.dtsi linux-5.10.y-b/arch/arm64/boot/dts/amlogic/meson-axg.dtsi
--- linux-5.10.y-a/arch/arm64/boot/dts/amlogic/meson-axg.dtsi	2022-12-20 13:57:13.000000000 +0000
+++ linux-5.10.y-b/arch/arm64/boot/dts/amlogic/meson-axg.dtsi	2023-01-04 09:28:42.000000000 +0000
@@ -1531,6 +1531,14 @@
 						bias-disable;
 					};
 				};
+
+				pwm_ao_c_ao8_pins:pwm_ao_c_ao8 {
+					mux {
+						groups = "pwm_ao_c_ao8";
+						function = "pwm_ao_c";
+						bias-disable;
+					};
+				};
 			};
 
 			sec_AO: ao-secure@140 {
@@ -1539,13 +1547,6 @@
 				amlogic,has-chip-id;
 			};
 
-			pwm_AO_cd: pwm@2000 {
-				compatible = "amlogic,meson-axg-ao-pwm";
-				reg = <0x0 0x02000  0x0 0x20>;
-				#pwm-cells = <3>;
-				status = "disabled";
-			};
-
 			uart_AO: serial@3000 {
 				compatible = "amlogic,meson-gx-uart", "amlogic,meson-ao-uart";
 				reg = <0x0 0x3000 0x0 0x18>;
@@ -1581,6 +1582,15 @@
 				status = "disabled";
 			};
 
+			pwm_AO_cd: pwm@2000 {
+				compatible = "amlogic,meson-axg-ao-pwm";
+				reg = <0x0 0x02000  0x0 0x20>;
+				#pwm-cells = <3>;
+				clocks = <&xtal>,<&xtal>,<&xtal>,<&xtal>;
+				clock-names = "clkin0","clkin1","clkin2","clkin3";
+				status = "okay";
+			};
+
 			ir: ir@8000 {
 				compatible = "amlogic,meson-gxbb-ir";
 				reg = <0x0 0x8000 0x0 0x20>;
@@ -1751,18 +1761,6 @@
 			#size-cells = <2>;
 			ranges = <0x0 0x0 0x0 0xffe00000 0x0 0x200000>;
 
-			sd_emmc_b: sd@5000 {
-				compatible = "amlogic,meson-axg-mmc";
-				reg = <0x0 0x5000 0x0 0x800>;
-				interrupts = <GIC_SPI 217 IRQ_TYPE_EDGE_RISING>;
-				status = "disabled";
-				clocks = <&clkc CLKID_SD_EMMC_B>,
-					<&clkc CLKID_SD_EMMC_B_CLK0>,
-					<&clkc CLKID_FCLK_DIV2>;
-				clock-names = "core", "clkin0", "clkin1";
-				resets = <&reset RESET_SD_EMMC_B>;
-			};
-
 			sd_emmc_c: mmc@7000 {
 				compatible = "amlogic,meson-axg-mmc";
 				reg = <0x0 0x7000 0x0 0x800>;
Binary files linux-5.10.y-a/arch/arm64/boot/dts/amlogic/._meson-axg-jethome-jethub-j100.dts and linux-5.10.y-b/arch/arm64/boot/dts/amlogic/._meson-axg-jethome-jethub-j100.dts differ
diff -urN linux-5.10.y-a/arch/arm64/boot/dts/amlogic/meson-axg-jethome-jethub-j100.dts linux-5.10.y-b/arch/arm64/boot/dts/amlogic/meson-axg-jethome-jethub-j100.dts
--- linux-5.10.y-a/arch/arm64/boot/dts/amlogic/meson-axg-jethome-jethub-j100.dts	2022-12-21 03:54:40.000000000 +0000
+++ linux-5.10.y-b/arch/arm64/boot/dts/amlogic/meson-axg-jethome-jethub-j100.dts	2023-01-11 03:43:43.000000000 +0000
@@ -18,23 +18,61 @@
 		device_type = "memory";
 		reg = <0x0 0x0 0x0 0x40000000>;
 	};
-};
 
+	aml_bt: aml_bt {
+		compatible = "amlogic, aml-bt";
+		status = "disabled";
+	};
+
+	aml_wifi: aml_wifi {
+		compatible = "amlogic, aml-wifi";
+		status = "okay";
+		irq_trigger_type = "GPIO_IRQ_LOW";
+		//dhd_static_buf; /* if use bcm wifi, config dhd_static_buf */
+		//pinctrl-0 = <&pwm_e_pins>;
+		//pinctrl-names = "default";
+		pwm_config = <&wifi_pwm_conf>;
+		//pwm_config = <&sdio_pwrseq>;
+	};
 
-/* wifi module */
-&sd_emmc_b {
-	non-removable;
-
-	brcmf: wifi@1 {
-		reg = <1>;
-		compatible = "brcm,bcm4329-fmac";
+	
+	wifi_pwm_conf:wifi_pwm_conf{
+		pwm_channel1_conf {
+			pwms = <&pwm_AO_cd 0 30541 0>;
+			duty-cycle = <15270>;
+			times = <10>;
+		};
+		
+		pwm_channel2_conf {
+			pwms = <&pwm_AO_cd 2 30500 0>;
+			duty-cycle = <15250>;
+			times = <12>;
+		};
 	};
+	
 };
 
 /* UART Bluetooth */
+/*
 &uart_B {
 	bluetooth {
 		compatible = "brcm,bcm43438-bt";
 		shutdown-gpios = <&gpio GPIOZ_7 GPIO_ACTIVE_HIGH>;
 	};
 };
+*/
+
+&aml_wifi{
+	status = "okay";
+	interrupt-gpios = <&gpio  GPIOX_6  GPIO_ACTIVE_HIGH>;
+	power_on-gpios = <&gpio   GPIOX_7  GPIO_ACTIVE_HIGH>;
+	//power_on_2-gpios= <&gpio  GPIOX_16 GPIO_ACTIVE_HIGH>;
+};
+
+&aml_bt {
+
+	status = "okay";
+	reset-gpios = <&gpio    GPIOX_21 GPIO_ACTIVE_HIGH>;
+	#btwakeup-gpios = <&gpio_ao GPIOAO_8 GPIO_ACTIVE_HIGH>;
+	#hostwake-gpios = <&gpio GPIOZ_6 GPIO_ACTIVE_HIGH>;
+};
\ No newline at end of file
Binary files linux-5.10.y-a/arch/arm64/boot/dts/amlogic/._meson-axg-jethome-jethub-j1xx.dtsi and linux-5.10.y-b/arch/arm64/boot/dts/amlogic/._meson-axg-jethome-jethub-j1xx.dtsi differ
diff -urN linux-5.10.y-a/arch/arm64/boot/dts/amlogic/meson-axg-jethome-jethub-j1xx.dtsi linux-5.10.y-b/arch/arm64/boot/dts/amlogic/meson-axg-jethome-jethub-j1xx.dtsi
--- linux-5.10.y-a/arch/arm64/boot/dts/amlogic/meson-axg-jethome-jethub-j1xx.dtsi	2022-12-21 03:54:40.000000000 +0000
+++ linux-5.10.y-b/arch/arm64/boot/dts/amlogic/meson-axg-jethome-jethub-j1xx.dtsi	2023-02-01 03:40:55.684128321 +0000
@@ -16,7 +16,8 @@
 / {
 	aliases {
 		serial0 = &uart_AO;   /* Console */
-		serial2 = &uart_AO_B; /* External UART (Wireless Module) */
+		serial2 = &uart_A; /* External UART (Wireless Module) */
+		serial3 = &uart_AO_B; /* ZigBee */
 		ethernet0 = &ethmac;
 	};
 
@@ -35,14 +36,7 @@
 		reset-gpios = <&gpio BOOT_9 GPIO_ACTIVE_LOW>;
 	};
 
-	vcc_3v3: regulator-vcc_3v3 {
-		compatible = "regulator-fixed";
-		regulator-name = "VCC_3V3";
-		regulator-min-microvolt = <3300000>;
-		regulator-max-microvolt = <3300000>;
-		vin-supply = <&vddao_3v3>;
-		regulator-always-on;
-	};
+	
 
 	vcc_5v: regulator-vcc_5v {
 		compatible = "regulator-fixed";
@@ -61,6 +55,15 @@
 		regulator-always-on;
 	};
 
+	vcc_3v3: regulator-vcc_3v3 {
+		compatible = "regulator-fixed";
+		regulator-name = "VCC_3V3";
+		regulator-min-microvolt = <3300000>;
+		regulator-max-microvolt = <3300000>;
+		vin-supply = <&vddao_3v3>;
+		regulator-always-on;
+	};
+
 	vddio_ao18: regulator-vddio_ao18 {
 		compatible = "regulator-fixed";
 		regulator-name = "VDDIO_AO18";
@@ -97,19 +100,24 @@
 		regulator-always-on;
 	};
 
+	/*
 	sdio_pwrseq: sdio-pwrseq {
 		compatible = "mmc-pwrseq-simple";
-		reset-gpios = <&gpio GPIOX_20 GPIO_ACTIVE_LOW>;
+		//reset-gpios = <&gpio GPIOX_7 GPIO_ACTIVE_LOW>;
 		clocks = <&wifi32k>;
 		clock-names = "ext_clock";
 	};
-
+	
 	wifi32k: wifi32k {
 		compatible = "pwm-clock";
 		#clock-cells = <0>;
 		clock-frequency = <32768>;
-		pwms = <&pwm_ab 0 30518 0>; /* PWM_A at 32.768KHz */
+		//pwms = <&pwm_AO_cd 0 30541 0>;
+		pwms = <&pwm_ab 0 30541 0>;
+		duty-cycle = <15270>;
+		times = <10>;
 	};
+	*/
 
 	thermal-zones {
 		cpu_thermal: cpu-thermal {
@@ -186,7 +194,7 @@
 };
 
 &ethmac {
-	status = "okay";
+	status = "disabled";
 	pinctrl-0 = <&eth_rmii_x_pins>;
 	pinctrl-names = "default";
 	phy-handle = <&eth_phy0>;
@@ -226,7 +234,7 @@
 
 /* Peripheral I2C bus (on motherboard) */
 &i2c_AO {
-	status = "okay";
+	status = "disabled";
 	pinctrl-0 = <&i2c_ao_sck_10_pins>, <&i2c_ao_sda_11_pins>;
 	pinctrl-names = "default";
 };
@@ -237,6 +245,33 @@
 	pinctrl-names = "default";
 };
 
+&pwm_AO_cd{
+	status = "okay";
+	pinctrl-0 = <&pwm_ao_c_ao8_pins>;
+	pinctrl-names = "default";
+};
+
+/* emmc storage */
+&sd_emmc_c {
+	status = "okay";
+	pinctrl-0 = <&emmc_pins>, <&emmc_ds_pins>;
+	pinctrl-1 = <&emmc_clk_gate_pins>;
+	pinctrl-names = "default", "clk-gate";
+
+	bus-width = <8>;
+	cap-mmc-highspeed;
+	max-frequency = <100000000>;
+	non-removable;
+	disable-wp;
+	mmc-ddr-1_8v;
+	mmc-hs200-1_8v;
+
+	mmc-pwrseq = <&emmc_pwrseq>;
+
+	vmmc-supply = <&vcc_3v3>;
+	vqmmc-supply = <&vccq_1v8>;
+};
+
 /* wifi module */
 &sd_emmc_b {
 	status = "okay";
@@ -249,36 +284,34 @@
 
 	bus-width = <4>;
 	cap-sd-highspeed;
+	
+	sd-uhs-sdr12;
+	sd-uhs-sdr25;
+	sd-uhs-sdr50;
+	//sd-uhs-sdr104;
+	
 	max-frequency = <50000000>;
 	disable-wp;
 
+	/*
 	mmc-pwrseq = <&sdio_pwrseq>;
+	*/
 
 	vmmc-supply = <&vddao_3v3>;
-	vqmmc-supply = <&vddio_boot>;
-};
-
-/* emmc storage */
-&sd_emmc_c {
-	status = "okay";
-	pinctrl-0 = <&emmc_pins>, <&emmc_ds_pins>;
-	pinctrl-1 = <&emmc_clk_gate_pins>;
-	pinctrl-names = "default", "clk-gate";
+	vqmmc-supply = <&vccq_1v8>;
 
-	bus-width = <8>;
-	cap-mmc-highspeed;
-	max-frequency = <200000000>;
 	non-removable;
 	disable-wp;
-	mmc-ddr-1_8v;
-	mmc-hs200-1_8v;
-
-	mmc-pwrseq = <&emmc_pwrseq>;
+	init_core_phase = <3>;
+	init_tx_phase = <0>;
+	no-mmc;
+	no-sd;
 
-	vmmc-supply = <&vcc_3v3>;
-	vqmmc-supply = <&vccq_1v8>;
+	keep-power-in-suspend;
+	mmc_debug_flag;
 };
 
+
 /* UART Bluetooth */
 &uart_B {
 	status = "okay";
@@ -301,6 +334,12 @@
 	pinctrl-names = "default";
 };
 
+&uart_A {
+	status = "okay";
+	pinctrl-0 = <&uart_a_pins>,<&uart_a_cts_rts_pins>;
+	pinctrl-names = "default";
+};
+
 &usb {
 	status = "okay";
 	phy-supply = <&usb_pwr>;
@@ -316,14 +355,14 @@
 	gpio-line-names =
 		"", "", "", "", "", // 0 - 4
 		"", "", "", "", "", // 5 - 9
-		"UserButton", "", "", "", "", // 10 - 14
+		"", "", "", "", "", // 10 - 14
 		"", "", "", "", "", // 15 - 19
 		"", "", "", "", "", // 20 - 24
-		"", "LedRed", "LedGreen", "Output3", "Output2", // 25 - 29
-		"Output1", "", "", "", "", // 30 - 34
-		"", "ZigBeeBOOT", "", "", "", // 35 - 39
-		"1Wire", "ZigBeeRESET", "", "Input4", "Input3", // 40 - 44
-		"Input2", "Input1", "", "", "", // 45 - 49
+		"", "", "", "", "", // 25 - 29
+		"", "", "", "", "", // 30 - 34
+		"", "", "", "", "", // 35 - 39
+		"", "", "", "", "", // 40 - 44
+		"", "", "", "", "", // 45 - 49
 		"", "", "", "", "", // 50 - 54
 		"", "", "", "", "", // 55 - 59
 		"", "", "", "", "", // 60 - 64
Binary files linux-5.10.y-a/arch/arm64/boot/dts/._.DS_Store and linux-5.10.y-b/arch/arm64/boot/dts/._.DS_Store differ
Binary files linux-5.10.y-a/arch/arm64/boot/dts/.DS_Store and linux-5.10.y-b/arch/arm64/boot/dts/.DS_Store differ
Binary files linux-5.10.y-a/arch/arm64/._.DS_Store and linux-5.10.y-b/arch/arm64/._.DS_Store differ
Binary files linux-5.10.y-a/arch/arm64/.DS_Store and linux-5.10.y-b/arch/arm64/.DS_Store differ
Binary files linux-5.10.y-a/arch/._.DS_Store and linux-5.10.y-b/arch/._.DS_Store differ
Binary files linux-5.10.y-a/arch/.DS_Store and linux-5.10.y-b/arch/.DS_Store differ
Binary files linux-5.10.y-a/._.DS_Store and linux-5.10.y-b/._.DS_Store differ
Binary files linux-5.10.y-a/.DS_Store and linux-5.10.y-b/.DS_Store differ
